include/swift/Option/FrontendOptions.td

```diff
@@ -167,6 +149,40 @@ def disable_serialization_nested_type_lookup_table :
 def debug_crash_Group : OptionGroup<"<automatic crashing options>">;
 class DebugCrashOpt : Group<debug_crash_Group>;
 
+
+// Flags that are saved into parseable interfaces
+let Flags = [FrontendOption, NoDriverOption, HelpHidden, ParseableInterfaceOption] in {
+
+def enable_objc_interop :
+  Flag<["-"], "enable-objc-interop">,
+  HelpText<"Enable Objective-C interop code generation and config directives">;
+
+def disable_objc_interop :
+  Flag<["-"], "disable-objc-interop">,
+  HelpText<"Disable Objective-C interop code generation and config directives">;
+
+def enable_objc_attr_requires_foundation_module :
+  Flag<["-"], "enable-objc-attr-requires-foundation-module">,
+  HelpText<"Enable requiring uses of @objc to require importing the "
+           "Foundation module">;
+
+def disable_objc_attr_requires_foundation_module :
+  Flag<["-"], "disable-objc-attr-requires-foundation-module">,
+  HelpText<"Disable requiring uses of @objc to require importing the "
+           "Foundation module">;
+
+def enable_resilience : Flag<["-"], "enable-resilience">,
+   HelpText<"Compile the module to export resilient interfaces for all "
+            "public declarations by default">;
+
+def enable_class_resilience : Flag<["-"], "enable-class-resilience">,
+   HelpText<"Enable resilient layout for classes containing resilient value types">;
+
+def read_type_info_path_EQ : Joined<["-"], "read-type-info-path=">,
+  HelpText<"Read legacy type layout from the given path">;
```

Comment:
This one can't go here. I'm not sure what we're going to do about legacy type info, but the argument is a path, which means it can't go into a swiftinterface.

cc @slavapestov